pex/cli/commands/lock.py

```diff
@@ -555,25 +563,41 @@ def _export(self):
                 downloaded_artifact.artifact.fingerprint
             )
 
-        with self.output(self.options) as output:
-            pins = fingerprints_by_pin.keys()  # type: Iterable[Pin]
-            if self.options.sort_by == ExportSortBy.PROJECT_NAME:
-                pins = sorted(pins, key=attrgetter("project_name.normalized"))
-            for pin in pins:
-                fingerprints = fingerprints_by_pin[pin]
-                output.write(
-                    "{project_name}=={version} \\\n"
-                    "  {hashes}\n".format(
-                        project_name=pin.project_name,
-                        version=pin.version.raw,
-                        hashes=" \\\n  ".join(
-                            "--hash={algorithm}:{hash}".format(
-                                algorithm=fingerprint.algorithm, hash=fingerprint.hash
-                            )
-                            for fingerprint in fingerprints
-                        ),
+        if self.options.format == ExportFormat.PIP:
+            with self.output(self.options) as output:
+                pins = fingerprints_by_pin.keys()  # type: Iterable[Pin]
+                if self.options.sort_by == ExportSortBy.PROJECT_NAME:
+                    pins = sorted(pins, key=attrgetter("project_name.normalized"))
+                for pin in pins:
+                    fingerprints = fingerprints_by_pin[pin]
+                    output.write(
+                        "{project_name}=={version} \\\n"
+                        "  {hashes}\n".format(
+                            project_name=pin.project_name,
+                            version=pin.version.raw,
+                            hashes=" \\\n  ".join(
+                                "--hash={algorithm}:{hash}".format(
+                                    algorithm=fingerprint.algorithm, hash=fingerprint.hash
+                                )
+                                for fingerprint in fingerprints
+                            ),
+                        )
                     )
-                )
+        else:
+            vertices = set()
+            edges = defaultdict(list)
+            for src_pin, dst_pins in resolved.adjacency_list.items():
+                dst_pin_strs = sorted(str(dst_pin) for dst_pin in dst_pins)
+                vertices.add(str(src_pin))
+                vertices.update(dst_pin_strs)
+                edges[str(src_pin)] = dst_pin_strs
+            graph = {
```

Comment:
The single target selected above on 508-510 is controlled by command line flags like --python, --python-path, --interpreter-constraint, --platform and --complete-platform, but none of these need be present either. It seems like some node to store information about the rendered graph context would make sense; i.e.: for a universal lock, this would give the context for environment marker activation / deactivation assuming you include activation information in the graph and for a multi-lock lock file, it also indicates which lock is being talked about (say the CPython 3.7 locked resolve from a lock file containing 3 locks, 1 for 3.7, 1 for 3.8 and 1 for 3.9.